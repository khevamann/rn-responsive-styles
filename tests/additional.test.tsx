import * as renderer from 'react-test-renderer'
import OverwriteStyle from './OverwriteStyle'
import * as React from 'react'
import { cleanup } from '@testing-library/react-native'
import { ScaledSize } from 'react-native'

afterEach(cleanup)

const mockDimensions = ({ width }: Pick<ScaledSize, 'width'>) => {
  jest.resetModules()
  jest.doMock('react-native/Libraries/Utilities/Dimensions', () => ({
    get: jest.fn().mockReturnValue({ width }),
    addEventListener: jest.fn(),
  }))
}

describe('additional tests for specific cases', () => {
  test('tests that regex only picks up the exact class specified', () => {
    mockDimensions({ width: 470 })
    const tree = renderer.create(<OverwriteStyle />).toJSON()
    expect(tree).toMatchSnapshot('small')
  })
  test('tests that regex does not include extra classes', () => {
    mockDimensions({ width: 770 })
    const tree = renderer.create(<OverwriteStyle />).toJSON()
    expect(tree).toMatchSnapshot('medium')
  })
})
