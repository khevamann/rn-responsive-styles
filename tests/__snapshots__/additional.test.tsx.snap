// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`additional tests for specific cases tests that regex does not include extra classes: medium 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 40,
      }
    }
  >
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;

exports[`additional tests for specific cases tests that regex only picks up the exact class specified: small 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 40,
      }
    }
  >
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;
